### PR TITLE
Fix warnings/issues reported in Windows and by CodeQL/Coverity

### PR DESCRIPTION
**Pull request by wenyongh**
Fix the warnings and issues reported:
- in Windows platform
- by CodeQL static code analyzing
- by Coverity static code analyzing

And update CodeQL script to build exception handling and memory features.

### CODE REVIEW

File: core/iwasm/common/wasm_memory.c
Comment (TianlongLiang) [inline core/iwasm/common/wasm_memory.c:1]: Maybe should also modify the implementation of `align_as_and_cast`?
Currently, it returns a `uint32`, maybe to something like:
```C
static uint64
align_as_and_cast(uint64 size, uint64 alignment)
{
    uint64 aligned_size = (size + alignment - 1) & ~(alignment - 1);

    return aligned_size;
}
```

Comment (wenyongh) [reply core/iwasm/common/wasm_memory.c:1]: Good catch! It is really an issue to truncate uint64 to uint32 and return uint32.

**Comment by TianlongLiang** [comment]
LGTM

**Comment by wenyongh** [comment]
Merge the PR since it was reviewed and tested.

**Pull request merged**